app/javascript/src/utils/compiler/compile.ts

```diff
@@ -28,10 +28,9 @@ export function compile(overrideContent = "", singleLanguageOverride = null): st
   joinedItems = evaluateForLoops(joinedItems)
   joinedItems = evaluateEachLoops(joinedItems)
   joinedItems = evaluateConditionals(joinedItems)
-  joinedItems = convertTranslations(joinedItems, singleLanguageOverride)
 
   const variables = compileVariables(joinedItems)
   const subroutines = compileSubroutines(joinedItems)
 
-  return settings + variables + subroutines + joinedItems
+  return convertTranslations(settings + variables + subroutines + joinedItems, singleLanguageOverride)
```

Comment:
Really only `settings` needs its translations converted. Maybe we can just call `convertTranslations()` in `joinedItems` and in `settings`? It'd technically be less work.

```ts
settings = convertTranslations(settings, singleLanguageOverride)

joinedItems = convertTranslations(joinedItems, singleLanguageOverride)

// ...

return settings + variables + subroutines + joinedItems
```

Reply:
Yes, but I'm choosing readability over speed here. Performance is not an concern when compiling.